src/Stratis.Features.FederatedPeg/TargetChain/CrossChainTransferStore.cs

```diff
@@ -1100,70 +1105,42 @@ private OutPoint EarliestOutput(Transaction transaction)
             return transaction.Inputs.Select(i => i.PrevOut).OrderBy(t => t, comparer).FirstOrDefault();
         }
 
-        private ICrossChainTransfer[] GetTransfersByStatusInternalLocked(CrossChainTransferStatus[] statuses, bool sort = false, bool validate = true)
-        {
-            var depositIds = new HashSet<uint256>();
-            foreach (CrossChainTransferStatus status in statuses)
-                depositIds.UnionWith(this.depositsIdsByStatus[status]);
-
-            uint256[] partialTransferHashes = depositIds.ToArray();
-            ICrossChainTransfer[] partialTransfers = this.Get(partialTransferHashes).Where(t => t != null).ToArray();
-
-            if (validate)
-            {
-                this.ValidateCrossChainTransfers(partialTransfers);
-                partialTransfers = partialTransfers.Where(t => statuses.Contains(t.Status)).ToArray();
-            }
-
-            if (!sort)
-            {
-                return partialTransfers;
-            }
-
-            // When sorting, Suspended transactions will have null PartialTransactions. Always put them last in the order they're in.
-            IEnumerable<ICrossChainTransfer> unsortable = partialTransfers.Where(x => x.Status == CrossChainTransferStatus.Suspended || x.Status == CrossChainTransferStatus.Rejected);
-            IEnumerable<ICrossChainTransfer> sortable = partialTransfers.Where(x => x.Status != CrossChainTransferStatus.Suspended && x.Status != CrossChainTransferStatus.Rejected);
-
-            return sortable.OrderBy(t => this.EarliestOutput(t.PartialTransaction), Comparer<OutPoint>.Create((x, y) =>
-                    ((FederationWalletManager)this.federationWalletManager).CompareOutpoints(x, y)))
-                .Concat(unsortable)
-                .ToArray();
-        }
-
         /// <inheritdoc />
-        public ICrossChainTransfer[] GetTransfersByStatus(CrossChainTransferStatus[] statuses, bool sort = false)
+        public ICrossChainTransfer[] GetTransfersByStatus(CrossChainTransferStatus[] statuses, bool sort = false, bool validate = true)
```

Comment:
@codingupastorm I still feel that it is wrong that the periodic loop calls "querying" methods that could potentially make changes to underlying state. As well as the fact the PL can make a call to the FMW which locks it. Feels a bit like we are mixing responsibility. Perhaps in another PR we can look at improving this?